### DISCUSSION TITLE
better support for non-standard tiff tags, eg. GeoTIFF

### DISCUSSION THREAD

**Discussion by petoor** [libvips/libvips#2377] [Ideas]
Hi John.

I have a geotiff image with metadata. I found that his example works as well [image_link](https://download.osgeo.org/geotiff/samples/GeogToWGS84GeoKey/GeogToWGS84GeoKey5.tif)

Example code.
```
import pyvips
from osgeo import gdal

geo = gdal.Open("GeogToWGS84GeoKey5.tif")
info = gdal.Info(geo, format="json")
info ... a lot of metadata ...

im = pyvips.Image.tiffload("GeogToWGS84GeoKey5.tif")
[field for field in im.get_fields()]
['width', 'height', 'bands', 'format', 'coding', 'interpretation', 'xoffset', 'yoffset', 'xres', 'yres', 'filename', 'vips-loader', 'n-pages', 'resolution-unit', 'orientation']
Does not contain a 'image-description' or metadata reference
```
So the question is. Is there a way to get vips to return geotiff metadata?

NB: Getting gdal can be a little tricky due to different versions not really talking together that well, but i found
apt install -y libgdal-dev
pip install pygdal==3.0.4.6
to work on ubuntu 20.04 and python3.6+

**Comment by jcupitt** [comment]
Hi @petoor,

geotiff implements metadata by adding extra, non-standard TIFF tags. You can see with `tiffinfo`:

```
$ tiffinfo GeogToWGS84GeoKey5.tif 
TIFFReadDirectory: Warning, Unknown field with tag 33550 (0x830e) encountered.
TIFFReadDirectory: Warning, Unknown field with tag 33922 (0x8482) encountered.
TIFFReadDirectory: Warning, Unknown field with tag 34735 (0x87af) encountered.
TIFFReadDirectory: Warning, Unknown field with tag 34736 (0x87b0) encountered.
...
```

libvips doesn't try to read unknown TIFF tags, though I suppose it could read ones with a known type. `tiffinfo` goes on to display:

```
  Tag 33550: 0.000028,0.000028,1.000000
  Tag 33922: 50.500000,50.500000,0.000000,9.001057,52.001376,0.000000
  Tag 34735: 1,1,0,7,1024,0,1,2,1025,0,1,1,2048,0,1,32767,2050,0,1,32767,2054,0,1,9102,2056,0,1,7004,2062,34736,7,0
  Tag 34736: 598.100000,73.700000,418.200000,0.202000,0.045000,-2.455000,6.700000
```

Which libvips could represent as int and double arrays. Let's tag this as an enhancement.

**Reply by petoor** [reply]
In my case the interesting part is the EPSG information (coordinateSystem). 
gdal.info shows something like this.
`{'description': 'GeogToWGS84GeoKey5.tif', 'driverShortName': 'GTiff', 'driverLongName': 'GeoTIFF', 'files': ['GeogToWGS84GeoKey5.tif'], 'size': [101, 101], 'coordinateSystem': {'wkt': 'BOUNDCRS[\n    SOURCECRS[\n  ... (pruned for readability)`

**Reply by jcupitt** [reply]
I doubt if all that could be supported, but the simple numeric arrays probably wouldn't be too hard.

GeoTIFF images are not standard TIFFs and you can't really use regular TIFF tools with them.

**Reply by petoor** [reply]
Sure, no problem, I can just use gdal.
It was more the notion of libvips is good for very large images, drone images are very larges images, so it would be nice with support for gdal metadata. I guess this would requre some kind of gdal plugin.
